Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/ExportImport2/PriceExportPagedDataSource.cs

```diff
@@ -23,26 +24,39 @@ public class PriceExportPagedDataSource : BaseExportPagedDataSource
         private readonly IPricingSearchService _searchService;
         private readonly IPricingService _pricingService;
         private readonly IItemService _itemService;
+        private readonly IAuthorizationService _authorizationService;
+        private readonly IUserClaimsPrincipalFactory<ApplicationUser> _userClaimsPrincipalFactory;
+        private readonly UserManager<ApplicationUser> _userManager;
 
         public PriceExportPagedDataSource(IPricingSearchService searchService,
             IPricingService pricingService,
             IItemService itemService,
-            IAuthorizationPolicyProvider authorizationPolicyProvider,
             IAuthorizationService authorizationService,
             IUserClaimsPrincipalFactory<ApplicationUser> userClaimsPrincipalFactory,
             UserManager<ApplicationUser> userManager)
-            : base(authorizationPolicyProvider, authorizationService, userClaimsPrincipalFactory, userManager)
         {
             _searchService = searchService;
             _pricingService = pricingService;
             _itemService = itemService;
+            _authorizationService = authorizationService;
+            _userClaimsPrincipalFactory = userClaimsPrincipalFactory;
+            _userManager = userManager;
         }
 
         protected override FetchResult FetchData(SearchCriteriaBase searchCriteria)
         {
             Price[] result;
             int totalCount;
 
+            var user = _userManager.FindByNameAsync(DataQuery.UserName).GetAwaiter().GetResult();
+            var claimsPrincipal = _userClaimsPrincipalFactory.CreateAsync(user).GetAwaiter().GetResult();
+            var authorizationResult = _authorizationService.AuthorizeAsync(claimsPrincipal, null, new PricingAuthorizationRequirement(ModuleConstants.Security.Permissions.Export)).GetAwaiter().GetResult();
```

Comment:
We could use AuthorizeAsync with specific policy name (it will be found using our implementation of IAuthorizationPolicyProvider):
```
            var authorizationResult = await _authorizationService.AuthorizeAsync(User, ModuleConstants.Security.Permissions.Read);

```

Reply:
Besides export check, we need to check Read permission when fetching data.
These comments relate to all export data sources.

Reply:
Also need to remove these attributes from data sources, as they were used for previous implementation:
    [Authorize(ModuleConstants.Security.Permissions.Export)]
    [Authorize(ModuleConstants.Security.Permissions.Read)]

Reply:
Also i'm thinking on creating some type with ability to check array of permissions, e.g.
```
        interface IPermissionAuthorizationCheckService
        {
            Task<AuthorizationResult> AuthorizeAsync(string userName, params string[] permissions);
        }

```
We could use it everywhere for our checks in pricing module. Please place it in VirtoCommerce.Platform.Security.Services